raiden-dapp/src/components/dialogs/ConnectionPendingDialog.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <raiden-dialog :visible="dialogVisible" :hide-close="true">
+    <v-card-title>
+      {{ $t('home.connection-pending-dialog.title') }}
+    </v-card-title>
+
+    <v-card-text>
+      <spinner :size="60" class="my-4" />
+      {{ $t('home.connection-pending-dialog.text') }}
+    </v-card-text>
+  </raiden-dialog>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+
+import RaidenDialog from '@/components/dialogs/RaidenDialog.vue';
+import Spinner from '@/components/icons/Spinner.vue';
+
+const SHOW_DIALOG_TIMEOUT = 7000;
+
+/*
+  Please note that because the underlying dialog component by Vuetify, it is not
+  possible to implement the delayed rendering with a Vue transition or pure CSS
+  effect. Therefore this alternative approach of using the life-cycle hook and
+  a timer is necessary.
+*/
+@Component({ components: { RaidenDialog, Spinner } })
+export default class ConnectionPendingDialog extends Vue {
+  dialogVisible = false;
+
+  mounted() {
+    this.delayVisibilityOfDialog();
+  }
+
+  delayVisibilityOfDialog(): void {
```

Comment:
Not sure I understand why we are waiting 7 seconds to display the dialog?

Reply:
The idea was to support the user when it might be necessary but don't annoy him. So theoretically if everything is fine the user should never see this dialog as it is now. But when after some time the connection process did not finish, it is likely that the user is missing to do something. Like we all know the case where MetaMask is not popping up and we are waiting for a long time. Or especially for WalletConnect where you manually need to open the app.
So the intention is too keep the user-flow nice and clean, but when it seems necessary support him.

Reply:
I will try it out first and see how it feels like with your current implementation.

Reply:
Hmm, so for me the QR code appeared and after I've logged in to WalletConnect and scanned the QR code the dialog and spinner was already visible. So I guess a user will always see it because in many cases it will take more than 7 seconds to pick up your phone and all that stuff?

Reply:
We had a discussion about this in a call. The result is that we are unsure about the whole flow in detail. Especially because with the upcoming connection manager it will change. We have arguments for and against different solutions here. As a conclusion we are in a pat situation. Therefore, for the moment, we keep it as what it was introduced for: a fix for the WalletConnect session issue. Once the new user flow with the connection manager got established, we can evaluate how well this fits. For example will this close Talel his comment that the dialog will be always visible due to the long session creation process. Maybe we will even merge the dialogs content into the new loading indicator as this will be target of change anyway.

As the comment might show, many things are in change and the vision is not 100% clear. Therefore we do one step after each other and improve other time.